Lib/test/test_posix.py

```diff
@@ -1622,6 +1622,24 @@ def test_setsigmask_wrong_type(self):
                             os.environ, setsigmask=[signal.NSIG,
                                                     signal.NSIG+1])
 
+    def test_start_new_session(self):
+        import subprocess
+        # For code coverage of calling setsid().  We don't care if we get an
+        # EPERM error from it depending on the test execution environment, that
+        # still indicates that it was called.
+        try:
+            output = subprocess.check_output(
+                    [sys.executable, "-c",
+                     "import os; print(os.getpgid(os.getpid()))"],
+                    start_new_session=True)
+        except OSError as e:
```

Comment:
I suggest to use "except PermissionError: pass" (written on two lines) instead.

Note: This exception also catch EACCES which is a good thing IMHO.